control/control.go

```diff
@@ -225,6 +239,83 @@ func New(cfg *Config) *pluginControl {
 	return c
 }
 
+func (p *pluginControl) HandleGomitEvent(e gomit.Event) {
+	switch v := e.Body.(type) {
+	case *control_event.LoadPluginEvent:
+		// Get all known tasks data
+		for taskID, taskData := range p.taskIDData {
+			// Expand every task metric namespaces (with "*") to corresponding metric calalog namespaces
+			for _, tm := range taskData.metrics {
+
+				matchedNss, err := p.metricCatalog.MatchNamespaces(tm.Namespace())
+
+				if err != nil {
+					log.WithFields(log.Fields{
+						"_block":         "control",
+						"event":          v.Namespace(),
+						"plugin-name":    v.Name,
+						"plugin-version": v.Version,
+						"plugin-type":    core.PluginType(v.Type).String(),
+					}).Error("error matching task namespace with metric catalog")
+				} else {
+					if len(matchedNss) > 0 {
+						depMts := []core.Metric{}
+						for _, ns := range matchedNss {
+
+							// Get expanded namespace data from metric catalog
+							if m, err := p.metricCatalog.Get(ns, tm.Version()); err == nil {
+								// Check if expanded namespace belongs to currently loaded plugin
+								if m.Plugin.TypeName() == core.PluginType(v.Type).String() && m.Plugin.Name() == v.Name {
+									depMts = append(depMts, &metric{
+										namespace: ns,
+										version:   tm.Version(),
+										config:    taskData.configTree.Get(ns.Strings()),
+									})
+								}
+							}
+						}
+
+						// Validate and subscribe deps for loaded plugin
+						errs := p.ValidateDeps(depMts, taskData.plugins)
+						if len(errs) > 0 {
+							log.WithFields(log.Fields{
+								"_block":         "control",
+								"event":          v.Namespace(),
+								"plugin-name":    v.Name,
+								"plugin-version": v.Version,
+								"plugin-type":    core.PluginType(v.Type).String(),
+							}).Error("error validating dependencies")
+						} else {
+							cps := returnCorePlugin(taskData.plugins)
+							errs = p.SubscribeDeps(taskID, depMts, cps)
```

Comment:
Please, correct me if I'm wrong, but in taskData.plugins there are only procesor and publisher plugins requested in task manifest (so, loading a new plugin does not change anything in context of processors/publishers). So, there is no need to (re)subscribe them.  As I know, there should be made only subscription of collector plugins - normally collector plugins are gathered based on metrics namespace in `SubscribeDeps` (see [snap/control/control.go#L757](https://github.com/intelsdi-x/snap/blob/master/control/control.go#L757)) but in this case the step of `gatherCollectors(mts)` could be skipped, so implementing such function for example, `SubcribeCollectorDeps(taskId string, collectorPlugins []core.Plugin)` which will do exactly the same as in lines [762-785](https://github.com/intelsdi-x/snap/blob/master/control/control.go#L762). So, that can be used here too to subscribe already loaded collector which metrics matches with requested query of metrics.   I'm also considering that all of that in `case *control_event.LoadPluginEvent` should be execute only when there is no pool for this plugin. Otherwise, when there is a pool for such plugin, but in different version, it should work as it's done before [see snap/control/runner.go#327](https://github.com/intelsdi-x/snap/blob/master/control/runner.go#L327). What do You think?